server/app/views/admin/questions/QuestionsListView.java

```diff
@@ -113,37 +115,82 @@ private DivTag renderSummary(ActiveAndDraftQuestions activeAndDraftQuestions) {
+                        .build())
+            .sorted(
+                Comparator.<QuestionCardData, Instant>comparing(
+                        card ->
+                            getDisplayQuestion(card).getLastModifiedTime().orElse(Instant.EPOCH))
+                    .reversed()
+                    .thenComparing(
+                        card ->
+                            getDisplayQuestion(card).getQuestionText().getDefault().toLowerCase()))
+            .collect(ImmutableList.toImmutableList());
+
+    for (QuestionCardData card : cards) {
       Pair<DivTag, ImmutableList<Modal>> rowAndModals =
-          renderQuestionRow(name, activeAndDraftQuestions, request);
+          renderQuestionCard(card, activeAndDraftQuestions, request);
       rows.add(rowAndModals.getLeft());
       modals.addAll(rowAndModals.getRight());
     }
     return Pair.of(div().with(rows.build()), modals.build());
   }
 
+  @AutoValue
+  abstract static class QuestionCardData {
+    abstract Optional<QuestionDefinition> activeQuestion();
+
+    abstract Optional<QuestionDefinition> draftQuestion();
+
+    static Builder builder() {
+      return new AutoValue_QuestionsListView_QuestionCardData.Builder();
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setActiveQuestion(Optional<QuestionDefinition> v);
+
+      abstract Builder setDraftQuestion(Optional<QuestionDefinition> v);
+
+      abstract QuestionCardData build();
+    }
+  }
+
   /**
-   * Renders a row in the question list. The row contains question text, help text, active and draft
-   * versions, list of programs questions is used in and buttons to edit the question.
+   * Renders a card in the question list. The card contains question text, help text, active and
+   * draft versions, list of programs questions is used in and buttons to edit the question.
    */
-  private Pair<DivTag, ImmutableList<Modal>> renderQuestionRow(
-      String questionName, ActiveAndDraftQuestions activeAndDraftQuestions, Http.Request request) {
-    Optional<QuestionDefinition> activeDefinition =
-        activeAndDraftQuestions.getActiveQuestionDefinition(questionName);
-    Optional<QuestionDefinition> draftDefinition =
-        activeAndDraftQuestions.getDraftQuestionDefinition(questionName);
-    if (draftDefinition.isEmpty() && activeDefinition.isEmpty()) {
+  private Pair<DivTag, ImmutableList<Modal>> renderQuestionCard(
+      QuestionCardData cardData,
+      ActiveAndDraftQuestions activeAndDraftQuestions,
+      Http.Request request) {
+    if (cardData.draftQuestion().isEmpty() && cardData.activeQuestion().isEmpty()) {
       throw new IllegalArgumentException("Did not receive a valid question.");
     }
-    QuestionDefinition latestDefinition = draftDefinition.orElseGet(activeDefinition::get);
+    QuestionDefinition latestDefinition =
+        cardData.draftQuestion().orElseGet(() -> cardData.activeQuestion().get());
```

Comment:
Hm. Java doesn't allow you to do `cardData.activeQuestion()::get` ?

Reply:
Switched to that. I just hadn't thought of doing it that way :)